src/components/Topics.jsx

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from "react";
+import { getTopics } from "../api";
+import { Link } from "react-router-dom";
+
+function Topics({ setTopicInput }) {
+  const [topics, setTopics] = useState([]);
+
+  useEffect(() => {
+    getTopics().then((topics) => {
+      setTopics(topics);
+    });
+  }, []);
+
+  function handleClick(topic) {
+    setTopicInput(topic);
+    const elem = document.activeElement;
```

Comment:
When using react we don't use `document.` because react uses a virtual DOM to take care of thing. Look at using `useRef`